ui/pages/institutional/custody/custody.js

```diff
@@ -0,0 +1,648 @@
+        );
+        setSelectedCustodianType(connectRequestValue.custodianType);
+        setSelectedCustodianName(connectRequestValue.custodianName);
+        setApiUrl(connectRequestValue.apiUrl);
+        connect();
+      }
+    };
+
+    // call the function
+    fetchConnectRequest()
+      // make sure to catch any error
+      .catch(console.error);
+  }, [dispatch, connect, currentChainId, mmiActions]);
+
+  useEffect(() => {
+    const handleNetworkChange = async () => {
+      if (!isNaN(chainId)) {
+        const jwt = currentJwt || jwtList[0];
+
+        if (jwt && jwt.length) {
+          setAccounts(
+            await getCustodianAccounts(
+              jwt,
+              apiUrl,
+              selectedCustodianType,
+              true,
+            ),
+          );
+        }
+      }
+    };
+
+    if (parseInt(chainId, 16) !== chainId) {
+      setChainId(parseInt(currentChainId, 16));
+      handleNetworkChange();
+    }
+  }, [
+    getCustodianAccounts,
+    apiUrl,
+    currentJwt,
+    jwtList,
+    selectedCustodianType,
+    currentChainId,
+    chainId,
+  ]);
+
+  const debouncedSearch = useMemo(() => {
+    debounce((input) => {
+      getCustodianAccountsByAddress(
+        currentJwt,
+        apiUrl,
+        input,
+        selectedCustodianType,
+        chainId,
+      );
+    }, 750);
+  }, [
+    getCustodianAccountsByAddress,
+    apiUrl,
+    chainId,
+    currentJwt,
+    selectedCustodianType,
+  ]);
+
+  const handleSearchDebounce = useCallback(
+    (e) => {
+      debouncedSearch(e.target.value);
+    },
+    [debouncedSearch],
+  );
+
+  const onSearchChange = (input) => {
+    handleSearchDebounce(input);
+  };
+
+  const cancelConnectCustodianToken = () => {
+    setSelectedCustodianName('');
+    setSelectedCustodianType('');
+    setSelectedCustodianImage(null);
+    setSelectedCustodianDisplayName('');
+    setApiUrl('');
+    setCurrentJwt('');
+    setConnectError('');
+    setSelectError('');
+  };
+
+  const setSelectAllAccounts = (e) => {
+    const allAccounts = {};
+
+    if (e.currentTarget.checked) {
+      accounts.forEach((account) => {
+        allAccounts[account.address] = {
+          name: account.name,
+          custodianDetails: account.custodianDetails,
+          labels: account.labels,
+          token: currentJwt,
+          apiUrl,
+          chainId: account.chainId,
+          custodyType: selectedCustodianType,
+          custodyName: selectedCustodianName,
+        };
+      });
+      setSelectedAccounts(allAccounts);
+    } else {
+      setSelectedAccounts({});
+    }
+  };
+
+  return (
+    <Box>
+      {connectError && (
+        <Text textAlign={TEXT_ALIGN.CENTER} marginTop={3} padding={[2, 7, 5]}>
+          {connectError}
+        </Text>
+      )}
+
+      {selectError && (
+        <Text textAlign={TEXT_ALIGN.CENTER} marginTop={3} padding={[2, 7, 5]}>
+          {selectError}
+        </Text>
+      )}
+
+      {renderCustodianButtons()}
```

Comment:
This function is called exactly once.... and we do not want these render methods as I mentioned before. Please use useMemo or just run the iterator every time like you are doing now without the render method.

Reply:
Added the useMemo 👍🏼